File: src/isolated_validation/commons-fileupload-python/src/test/org/apache/commons/fileupload/DiskFileItemSerializeTest.py
from __future__ import annotations
import re
import tempfile
from pickle import Pickler, Unpickler
import unittest
import pytest
import pathlib
import io
from io import BytesIO
import typing
from typing import *
import os


class DiskFileItemSerializeTest:

    __threshold: int = 16
    __textContentType: str = "text/plain"
    __REPO: pathlib.Path = pathlib.Path(
        os.getenv("TEMP") or tempfile.gettempdir(), "diskfileitemrepo"
    )

    def __deserialize(self, baos: typing.Union[io.BytesIO, bytearray]) -> typing.Any:

        result = None
        byte_stream = BytesIO(baos) if isinstance(baos, bytearray) else baos
        try:
            ois = Unpickler(byte_stream)
            result = ois.load()
        finally:
            if isinstance(baos, bytearray):
                byte_stream.close()
        return result

    def __serialize(self, target: typing.Any) -> typing.Union[io.BytesIO, bytearray]:

        baos = BytesIO()
        oos = Pickler(baos)
        oos.dump(target)
        byte_array = baos.getvalue()
        baos.close()
        return byte_array

    def __createContentBytes(self, size: int) -> typing.List[int]:

        buffer = [str(i % 10) for i in range(size)]
        return [ord(c) for c in "".join(buffer)]

    def __compareBytes(
        self, text: str, origBytes: typing.List[int], newBytes: typing.List[int]
    ) -> None:

        assert origBytes is not None, "origBytes must not be null"
        assert newBytes is not None, "newBytes must not be null"
        assert len(origBytes) == len(newBytes), f"{text} byte[] length"

        for i in range(len(origBytes)):
            assert origBytes[i] == newBytes[i], f"{text} byte[{i}]"

File: java_projects/cleaned_final_projects/commons-fileupload/src/test/java/org/apache/commons/fileupload/DiskFileItemSerializeTest.java
/*
 * Licensed to the Apache Software Foundation (ASF) under one or more
 * contributor license agreements.  See the NOTICE file distributed with
 * this work for additional information regarding copyright ownership.
 * The ASF licenses this file to You under the Apache License, Version 2.0
 * (the "License"); you may not use this file except in compliance with
 * the License.  You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */
package org.apache.commons.fileupload;

import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertNotNull;

import java.io.ByteArrayInputStream;
import java.io.ByteArrayOutputStream;
import java.io.File;
import java.io.ObjectInputStream;
import java.io.ObjectOutputStream;

/** Serialization Unit tests for {@link org.apache.commons.fileupload.disk.DiskFileItem}. */
public class DiskFileItemSerializeTest {

    private static final File REPO =
            new File(System.getProperty("java.io.tmpdir"), "diskfileitemrepo");

    /** Content type for regular form items. */
    private static final String textContentType = "text/plain";

    /** Very low threshold for testing memory versus disk options. */
    private static final int threshold = 16;

    /** Helper method to test creation of a field when a repository is used. */

    /** Helper method to test creation of a field. */

    /**
     * Test creation of a field for which the amount of data falls below the configured threshold.
     */

    /** Test creation of a field for which the amount of data equals the configured threshold. */

    /**
     * Test creation of a field for which the amount of data falls above the configured threshold.
     */

    /** Test serialization and deserialization when repository is not null. */

    /** Test deserialization fails when repository is not valid. */

    /** Test deserialization fails when repository contains a null character. */

    /** Compare content bytes. */
    private void compareBytes(String text, byte[] origBytes, byte[] newBytes) {
        assertNotNull("origBytes must not be null", origBytes);
        assertNotNull("newBytes must not be null", newBytes);
        assertEquals(text + " byte[] length", origBytes.length, newBytes.length);
        for (int i = 0; i < origBytes.length; i++) {
            assertEquals(text + " byte[" + i + "]", origBytes[i], newBytes[i]);
        }
    }

    /** Create content bytes of a specified size. */
    private byte[] createContentBytes(int size) {
        StringBuilder buffer = new StringBuilder(size);
        byte count = 0;
        for (int i = 0; i < size; i++) {
            buffer.append(count + "");
            count++;
            if (count > 9) {
                count = 0;
            }
        }
        return buffer.toString().getBytes();
    }

    /** Create a FileItem with the specfied content bytes and repository. */

    /** Create a FileItem with the specfied content bytes. */

    /** Do serialization */
    private ByteArrayOutputStream serialize(Object target) throws Exception {
        ByteArrayOutputStream baos = new ByteArrayOutputStream();
        ObjectOutputStream oos = new ObjectOutputStream(baos);
        oos.writeObject(target);
        oos.flush();
        oos.close();
        return baos;
    }

    /** Do deserialization */
    private Object deserialize(ByteArrayOutputStream baos) throws Exception {
        Object result = null;
        ByteArrayInputStream bais = new ByteArrayInputStream(baos.toByteArray());
        ObjectInputStream ois = new ObjectInputStream(bais);
        result = ois.readObject();
        bais.close();

        return result;
    }
}
